plugin.video.themoviedb.helper/resources/lib/script.py

```diff
@@ -0,0 +1,490 @@
+            xbmc.executebuiltin('ActivateWindow(videos, {0}, return)'.format(self.params.get('call_path')))
+        elif self.params.get('call_update'):
+            xbmc.executebuiltin('Container.Update({0})'.format(self.params.get('call_update')))
+
+    def add_path(self):
+        url = self.params.get('add_path', '')
+        url = url.replace('info=play', 'info=details')
+        url = url.replace('info=seasons', 'info=details')
+        url = '{0}&{1}'.format(url, 'extended=True') if 'extended=True' not in url else url
+        if url == self.home.getProperty(self.prefixcurrent):
+            return  # Already added so let's quit as user probably clicked twice
+        self.position = self.get_position() + 1
+        self.set_props(self.position, url)
+        self.lock_path(self.params.get('prevent_del'))
+        self.call_auto()
+
+    def add_query(self):
+        with utils.busy_dialog():
+            query = utils.try_decode_string(self.params.get('add_query', ''))
+            item = utils.dialog_select_item(query)
+            if not item:
+                return
+            tmdb_id = self.tmdb.get_tmdb_id(self.params.get('type'), query=item, selectdialog=True)
+            if not tmdb_id:
+                utils.kodi_log(u'Unable to find TMDb ID!\nQuery: {0} Type: {1}'.format(self.params.get('add_query'), self.params.get('type')), 1)
+                return
+            url = 'plugin://plugin.video.themoviedb.helper/?info=details&amp;type={0}&amp;tmdb_id={1}'.format(self.params.get('type'), tmdb_id)
+            if url == self.home.getProperty(self.prefixcurrent):
+                return  # Already added so let's quit as user probably clicked twice
+            self.position = self.get_position() + 1
+            self.set_props(self.position, url)
+            self.lock_path(self.params.get('prevent_del'))
+        self.call_auto()
+
+    def add_prop(self):
+        item = utils.dialog_select_item(self.params.get('add_prop'))
+        if not item:
+            return
+        prop_name = '{0}{1}'.format(self.prefixname, self.params.get('prop_id'))
+        self.home.setProperty(prop_name, item)
+        self.call_auto()
+
+    def prev_path(self):
+        # Get current position and clear it
+        self.position = self.get_position()
+        self.home.clearProperty('{0}{1}'.format(self.prefixpath, self.position))
+
+        # If it was first position then let's clear everything
+        if not self.position > 1:
+            self.reset_props()
+            return
+
+        # Otherwise set previous position to current position
+        self.position -= 1
+        path = self.home.getProperty('{0}{1}'.format(self.prefixpath, self.position))
+        self.set_props(self.position, path)
+
+    def del_path(self):
+        if self.home.getProperty(self.prefixlock):
+            self.added_path = None
+            self.unlock_path()
+            return
+
+        self.prev_path()
+        self.call_window()
+
+    def play(self):
```

Comment:
same here. When is this called? For which type of items? Calling videodb:// or plugin:// paths with this method is fine. Getting remote streams and request playback via this method is not okay

Reply:
> Calling videodb:// or plugin:// paths with this method is fine.

That's exactly what it is for.  User can define player files for other addons e.g. Netflix  with defined rules for how to play the item.  When the user clicks an item to play in TMDbHelper, the plugin uses JSONRPC to walk the paths of the external addon to find a playable item according to matching rules. The plugin file path is then passed to the player and combined with the details grabbed from online.

The main reason this method is necessary is because some addons provide playable items without detailed information. Passing to Player() is the only way to combine a playable path with listitem details. Also there are some scenarios where passing the plugin path to  `PlayMedia()` causes a crash.

TMDbHelper is not providing any playable content - it is just providing a way to interface with other addons.